### PR TITLE
[snapshots] Refactor snapshot rollbacks in BaseVM

### PR DESCRIPTION
**Pull request by ricab**
Needs #3096 to be merged first.

### CODE REVIEW

File: src/platform/backends/shared/base_virtual_machine.h [hunk 1]
```diff
@@ -70,18 +70,33 @@ class BaseVirtualMachine : public VirtualMachine
...
+    using SnapshotMap = std::unordered_map<std::string, std::shared_ptr<Snapshot>>;
+
     template <typename LockT>
     void log_latest_snapshot(LockT lock) const;
+
     void load_generic_snapshot_info(const QDir& snapshot_dir);
     void load_snapshot_from_file(const QString& filename);
     void load_snapshot(const QJsonObject& json);
+
+    auto make_take_snapshot_rollback(std::unordered_map<std::string, std::shared_ptr<Snapshot>>::iterator it);
```
Comment (sharder996) [inline src/platform/backends/shared/base_virtual_machine.h:82]:
```suggestion
    auto make_take_snapshot_rollback(SnapshotMap::iterator it);
```

**Review by sharder996 — CHANGES_REQUESTED** [review]
Looks good, just one thing that should be changed.

btw, "guts" sounds so graphic :nauseated_face: :laughing:

**Comment by ricab** [comment]
Thanks @sharder996. I will change guts too. Suggestions welcome!

**Comment by sharder996** [comment]
Just messing with you @ricab! Though, if I was writing a similar function, I would have probably called it something like `take_snapshot_rollback_helper` :shrug:

**Comment by ricab** [comment]
Sure, no worries! Let's make it less noticeable.

**Review by sharder996 — APPROVED** [review]
Thanks, @ricab!

**Pull request merged**